Makefile

```diff
@@ -136,9 +137,75 @@ build-darwin:
 	  $(MAKE) build
 
 build-linux:
-	SC_GOOS=linux SC_GOARCH=amd64 BINDIR=bin/darwin_amd64 DOCKER=1 \
+	SC_GOOS=linux SC_GOARCH=amd64 BINDIR=bin/linux_amd64 DOCKER=1 \
 	  $(MAKE) build
 
+# Building Docker Images for our executables
+############################################
+images: catalog-image registry-image k8s-broker-image user-broker-image \
+  controller-image
+
+catalog-image: pkg/controller/catalog/Dockerfile build-linux
```

Comment:
This will still build all the binaries, even though it only builds one of the images, right? Can we make it just build the one it needs?